hack/determine_install_upgrade_version_test.py

```diff
@@ -0,0 +1,41 @@
+import unittest
+import rpmUtils.miscutils as rpmutils
+from determine_install_upgrade_version import *
+
+class TestPackage(object):
+    def __init__(self, version, release, vra):
+        self.version = version
+        self.release = release
+        self.vra = vra
+
+class DetermineInstallUpgradeVersionTestCase(unittest.TestCase):
+	"Test for `determine_install_upgrade_version.py`"
+	PKG_NAME = "origin"
+	MATCHING_1_5 = ["1.5.0-0.4.el7.x86_64"]
+	MATCHING_1_4 = ["1.4.1-1.el7.x86_64"]
+	MATCHING_1_3 = ["1.3.1-1.el7.x86_64", "1.3.2-3.el7.x86_64"]
+	MATCHING_1_2 = ["1.2.0-0.el7.x86_64", "1.2.5-0.el7.x86_64"]
+
+	def test_get_matching_versions(self):
+		self.assertEqual(get_matching_versions(self.PKG_NAME, test_pkgs_objs, "1.5"), self.MATCHING_1_5)
```

Comment:
Let's break these up into specific tests and name them by the intent that they surface. For instance, this test could be:

```py
def test_get_matching_versions_only_prelease(self):
    """ when only pre-release versions exist, they are returned """
    self.assertEqual(
        get_matching_versions('origin', ['1.5.0-0.4.el7.x86_64'], '1.5'),
        ['1.5.0-0.4.el7.x86_64']
    )
```

Then you could pair it with:
```py
def test_get_matching_versions_release_and_prelease(self):
    """ when both release and pre-release versions exist, only release versions are returned """
    self.assertEqual(
        get_matching_versions('origin', ['1.5.0-0.4.el7.x86_64','1.5.0-1.4.el7.x86_64'], '1.5'),
        ['1.5.0-1.4.el7.x86_64']
    )
```

And follow up with the other tests you have written -- when more than one version exists, all are returned, etc.